flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -91,6 +92,13 @@
      */
     private final Map<String, State<T>> states;
 
+    /**
+     * The lengths of a windowed pattern, as specified using the {@link
+     * org.apache.flink.cep.pattern.Pattern#within(Time, Pattern.WithinType)} Pattern.within(Time,
+     * WithinType)} method with {@code WithinType.PREVIOUS_AND_CURRENT}.
+     */
+    private final Map<String, Long> windowTimes;
```

Comment:
Do you think it makes sense to move the windowTime between previous and current event to *State*?

Reply:
@dianfu, IMO, the windowTime between previous and current event doesn't need to move to the `State`. Because this windowTime is the timeout condition of the NFA state, which state doesn't need to include the condition.